docs/contribute/framework.md

```diff
@@ -0,0 +1,145 @@
+---
+title: 'Contributing a Storybook framework'
+---
+
+A Storybook framework is a node package that enables out-of-the-box support for either a metaframework (like Next.js, NuxtJS, or SvelteKit) or a combination of a builder (Webpack or Vite) and a renderer (React, Angular, Vue, web components, etc). When the framework provides support for a metaframework, it also takes care of any additional configuration necessary to make Storybook behave as similarly as possible to apps generated by the metaframework. For example, `@storybook/nextjs` [recreates or mocks a number of features of Next.js apps](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/README.md#supported-features) inside Storybook.
+
+For your reference, you can view [all of the official Storybook frameworks](https://github.com/storybookjs/storybook/tree/next/code/frameworks), including their full source code and documentation.
+
+## How to make a framework
+
+### 1. Decide on a package name
+
+The name should start with `storybook-framework-` and then correspond to what your framework supports. For example, a framework targeting SvelteKit would be `storybook-framework-svelte-kit` and a framework targeting Stencil with Vite would be `storybook-framework-stencil-vite`. When not targeting a metaframework, the naming convention is `storybook-framework-<renderer>-<builder>`.
+
+### 2. Consider what your framework will need to do
+
+Remember, the goal is to make Storybook behave—out-of-the-box—as similarly as possible to the metaframework or builder-renderer combination you’re targeting. In particular for metaframeworks, this means attempting to recreate any builder or babel configuration provided by the metaframework. And you should try to do so in a way that respects existing project configuration as much as possible.
+
+The library or libraries your framework supports may have different major versions available. Carefully consider which versions of each library your framework will support. You will need to account for the changes within those different versions or split your framework into different versions/packages itself to support each library version. To help ongoing maintenance, please consider adding integration tests for the various library versions your framework supports.
+
+### 3. Write the README
+
+Before writing any code, we advise you to write up a helpful README. It should, at minimum, contain installation instructions and a list of available features. Use the [README for `@storybook/nextjs`](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/README.md) as a template. By writing the documentation first, it becomes a guide for your other work.
```

Comment:
@kylegach we can probably rename this  "Write documentation". What do you think?